### DISCUSSION TITLE
Query millions of rows of data at once.

### DISCUSSION THREAD

**Discussion by ChenpiDog** [dbeaver/dbeaver#22084] [General]
When querying millions of rows of data at a time, DBeaver queries are very slow and may eventually crash. For the same query, Navicat will be much faster and will not crash.

**Comment by LonwoLonwo** [comment]
Hello @ChenpiDog 

What database do you use? And what kind of query?

**Reply by ChenpiDog** [reply]
Greenplum

**Reply by ChenpiDog** [reply]
A simple Select query with filter conditions of '>=' and 'in' and one million rows of data.

**Reply by LonwoLonwo** [reply]
We have a bunch of settings for partial data reading.

You can change it here:

![2023-12-04 18_00_50-Preferences](https://github.com/dbeaver/dbeaver/assets/45152336/8484e2e0-7f0d-4312-9e20-1ca75543b003)
